### ISSUE TITLE
new rule `property-value-no-negative`

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#1003]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Throw error if find negative value with property, which not support negative value. Most often this means typing error.

> Which rule, if any, is this issue related to?

`property-value-no-negative`

> What CSS is needed to reproduce this issue?

```css
.foo {
  width: -100px
  height: -100px
  padding: -10px
  ...
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "property-value-no-negative": true
  }
}
```

> Which version of stylelint are you using?

e.g. `5.2.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

not matter

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

e.g. "Yes, it's related to SCSS maps."

not matter

> What actually happened (e.g. what warnings or errors you are getting)?

Nothing

**Comment by davidtheclark** [comment]
Thanks @evilebottnawi. In the future, feel free to delete the questions from the template that do not apply to your issue.

Do you have a list in mind of properties that do not support negative number values?

**Comment by jeddy3** [comment]
Would we be better helping out with [`css-values`](https://github.com/ben-eb/css-values) and then creating a more generic `declaration-no-invalid` rule, rather than something specific just to negative numbers.

The existing `value-no-*` rules (`value-no-vendor-prefix` and `value-no-longhard`) enforce a *preference* i.e. both prefixed and unprefixed values, as well as longhand and shorthand value are *valid*. So, they feel complimentary to a future `declaration-no-invalid` rule. Whereas this proposal feels like it'll become superfluous going forward.

What do you think?

**Comment by davidtheclark** [comment]
> Would we be better helping out with css-values and then creating a more generic declaration-no-invalid rule, rather than something specific just to negative numbers.

That would probably be the best long term approach. I have to have time to really dive into that project so don't know how far off the dream is :)

**Comment by evilebottnawi** [comment]
@jeddy3 sounds good, i think it would be a good idea first to create separate rules to understand how all these rules apply to write good css. In the future, already combine these rules. What do you think?

**Comment by jeddy3** [comment]
Closing this in favour of spending time focusing on [accepted proposals](https://github.com/stylelint/stylelint/issues?q=is%3Aissue+is%3Aopen+label%3A%22type%3A+rule%22) and [shoring up tests](https://github.com/stylelint/stylelint/issues?q=is%3Aissue+is%3Aopen+label%3A%22type%3A+tests%22).

Free free to experiment with this as a plugin though.
